### PR TITLE
Pre-check before querying with parameters & temporarily fixed bug from endpoint when creating a server 

### PR DESCRIPTION
**Pull request by nvthongswansea**

### CODE REVIEW

**Comment by itakouna** [comment]
@nvthongswansea I donot think precheck `id==""` will make a big this difference. I would suggest checking whether  `id` is a valid UUID.

**Comment by itakouna** [comment]
Plz, add a unit test for this case!

**Comment by nvthongswansea** [comment]
> Plz, add a unit test for this case!

Which one please?

**Comment by nvthongswansea** [comment]
> @nvthongswansea I don't think precheck `id==""` will make a big this difference. I would suggest checking whether `id` is a valid UUID.

It will do. As if you enter an empty `id`, the single object getters will act the same as the list getters do. For example `client.GetServer("")` performs exactly the same as `client.GetServerList()`. I discovered this problem during working with our terraform provider and this caused so much headache. Futhermore, we reduce a number of empty-uuid requests to the server (which is good).

**Comment by itakouna** [comment]
Nice. I forgot the string concatenation with ID.
What do you think about checking the validity of UUID format?https://godoc.org/github.com/google/uuid#Parse

**Comment by itakouna** [comment]
> > Plz, add a unit test for this case!
> 
> Which one please?

example:

```
func TestClient_GetServerEmptyString(t *testing.T) {
	server, client, mux := setupTestClient()
	defer server.Close()
	uri := path.Join(apiServerBase, dummyUUID)
	mux.HandleFunc(uri, func(writer http.ResponseWriter, request *http.Request) {
		assert.Equal(t, http.MethodGet, request.Method)
		fmt.Fprintf(writer, prepareServerHTTPGet(true))
	})
	res, err := client.GetServer("")

}```

**Comment by nvthongswansea** [comment]
> Nice. I forgot the string concatenation with ID.
> What do you think about checking the validity of UUID format?https://godoc.org/github.com/google/uuid#Parse

It depends on how sure UUID format isn't going to be changed in the future. If the format is solid (and the same as the format in the lib you sent me), it is ok to validate the UUIDs before sending any requests.

**Comment by itakouna** [comment]
> > Nice. I forgot the string concatenation with ID.
> > What do you think about checking the validity of UUID format?https://godoc.org/github.com/google/uuid#Parse
> 
> It depends on how sure UUID format isn't going to be changed in the future. If the format is solid (and the same as the format in the lib you sent me), it is ok to validate the UUIDs before sending any requests.

Ok. let's keep it as is.

File: objectstorage.go [hunk 1]
```diff
@@ -66,9 +67,12 @@ func (c *Client) GetObjectStorageAccessKeyList() ([]ObjectStorageAccessKey, erro
 }
 
 //GetObjectStorageAccessKey gets a specific object storage access key based on given id
-func (c *Client) GetObjectStorageAccessKey(id string) (ObjectStorageAccessKey, error) {
+func (c *Client) GetObjectStorageAccessKey(accessKey string) (ObjectStorageAccessKey, error) {
```
Comment (itakouna) [inline objectstorage.go]: In API docs, the name of the object's ID might change based on the resource name e.g,(iso_image_uuid, server_uuid, access_keys). I would keep `id` to keep consistency in gs-client.

File: common.go [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package gsclient
+
+import "github.com/google/uuid"
+
+//isValidUUID validates the uuid
+func isValidUUID(u string) bool {
+	_, err := uuid.Parse(u)
+	return err == nil
+}
```
Comment (itakouna) [inline common.go]: new line is missing!!

File: objectstorage.go (hunk 1 above)
Comment (nvthongswansea) [reply objectstorage.go]: But you can only use access key to access an object storage.

File: common_test.go [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+import (
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func Test_isValidUUID(t *testing.T) {
+	validUUIDTest := make([]uuidTestCase, len(uuidCommonTestCases))
+	copy(validUUIDTest, uuidCommonTestCases)
+	validUUIDTest = append(validUUIDTest, uuidTestCase{
```
Comment (itakouna) [inline common_test.go]: This name `validUUIDTest` is misleading. I expect a valid UUID to be tested, however, 'abc-123` is invalid. I would add two tests `validUUIDTest` and `invalidUUIDTest`.

File: objectstorage.go (hunk 1 above)
Comment (itakouna) [reply objectstorage.go]: `accessKey` is ID of firewall. All `Get` a resouce should use `id`.

Comment (nvthongswansea) [reply objectstorage.go]: > `accessKey` is ID of firewall. All `Get` a resouce should use `id`.

I couldn't find any information on our API docs about "`accessKey` is ID of firewall".

**Pull request merged**